Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -291,6 +319,7 @@ Connection::Connection(Identifier identifier, bool isServer, Client& client)
     , m_uniqueID(UniqueID::generate())
     , m_isServer(isServer)
     , m_connectionQueue(WorkQueue::create("com.apple.IPC.ReceiveQueue"))
+    , m_syncState(SyncMessageStateImpl::forCurrentThread(&RunLoop::main()))
```

Comment:
the sync state association must happen on open().

E.g. the connection might be created on main run loop, but the open says "deliver to this message receiver." So here you then want to change the api contract to "deliver to this message receiver in this runloop".